### PR TITLE
Enable CoreML EP for minimal extended mode

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: Enable CoreML EP for minimal extended mode

**Motivation and Context**
- There may be user case for iOS ORT mobile using CoreML EP
- Added UT for using coreml ep in minimal build
- Add missing 3rd party notice for coremltools
- Not included in this PR, CI change to run CoreML EP using ORT minimal and minimal ORT document updates

### CODE REVIEW

File: onnxruntime/test/providers/coreml/coreml_basic_test.cc [hunk 1]
```diff
@@ -13,7 +15,10 @@
 #include "test/util/include/test/test_environment.h"
 #include "test/util/include/test_utils.h"
 
+#if !defined(ORT_MINIMAL_BUILD)
+// if this is a full build we need the provider test utils
 #include "test/providers/provider_test_utils.h"
+#endif  // !(ORT_MINIMAL_BUILD
```
Comment (YUNQIUGUO) [inline onnxruntime/test/providers/coreml/coreml_basic_test.cc:21]: nit: // !(ORT_MINIMAL_BUILD)

**Review by YUNQIUGUO — APPROVED** [review]

File: onnxruntime/test/providers/coreml/coreml_basic_test.cc [hunk 2]
```diff
@@ -25,6 +30,11 @@ using namespace ::onnxruntime::logging;
 namespace onnxruntime {
 namespace test {
 
+// We want to run UT on CPU only to get output value without losing precision to pass the verification
+static uint32_t s_coreml_flags = COREML_FLAG_USE_CPU_ONLY;
```
Comment (edgchen1) [inline onnxruntime/test/providers/coreml/coreml_basic_test.cc:34]: const or constexpr

File: onnxruntime/test/providers/coreml/coreml_basic_test.cc [hunk 3]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
```
Comment (edgchen1) [inline onnxruntime/test/providers/coreml/coreml_basic_test.cc:4]: is this necessary? looks like this file is only included by cmake if onnxruntime_USE_COREML is set, and there's a new check in this PR:
```
if (onnxruntime_USE_COREML)
  if (onnxruntime_MINIMAL_BUILD AND NOT onnxruntime_EXTENDED_MINIMAL_BUILD)
    message(FATAL_ERROR "CoreML EP can not be used in a basic minimal build. Please build with '--minimal_build extended'")
  endif()
  ...
```

Comment (gwang-msft) [reply onnxruntime/test/providers/coreml/coreml_basic_test.cc:4]: Yes, I think we can remove this, maybe the one in nnapi_basic_test.cc as well

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**